backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionServiceImpl.java

```diff
@@ -87,23 +105,21 @@ public void createDatasetVersion(
             Any.pack(CreateDatasetVersion.Response.getDefaultInstance()));
       }
 
-      roleService.validateEntityUserWithUserInfo(
-          ModelDBServiceResourceTypes.DATASET,
-          request.getDatasetId(),
-          ModelDBServiceActions.UPDATE);
-
-      Dataset dataset = datasetDAO.getDatasetById(request.getDatasetId());
-      if (dataset.getDatasetType() != request.getDatasetType()) {
-        logAndThrowError(
-            ModelDBMessages.DATASET_VERSION_TYPE_NOT_MATCH_WITH_DATSET_TYPE,
-            Code.INVALID_ARGUMENT_VALUE,
-            Any.pack(CreateDatasetVersion.Response.getDefaultInstance()));
-      }
-
       /*Get the user info from the Context*/
       UserInfo userInfo = authService.getCurrentLoginUserInfo();
       DatasetVersion datasetVersion =
-          datasetVersionDAO.createDatasetVersion(request, dataset, userInfo);
+          datasetVersionDAO.getDatasetVersionFromRequest(authService, request, userInfo);
+      RepositoryIdentification repositoryIdentification =
+          RepositoryIdentification.newBuilder()
+              .setRepoId(Long.parseLong(request.getDatasetId()))
+              .build();
+      RepositoryFunction repositoryFunction =
+          (session) -> repositoryDAO.getRepositoryById(session, repositoryIdentification, true);
+      CreateCommitRequest.Response createCommitResponse =
+          commitDAO.setCommitFromDatasetVersion(
+              datasetVersion, blobDAO, metadataDAO, repositoryFunction);
+      datasetVersion =
```

Comment:
this seems wrong since the Dataset info is read from request and directly added to response, we should update response based on the blob which was written.